internal/charmstore/addentity.go

```diff
@@ -225,65 +241,107 @@ func (s *Store) addEntityFromReader(id *router.ResolvedURL, r io.ReadSeeker, blo
 	return nil
 }
 
-type preV5CompatibilityHackBlobInfo struct {
+type compatibilityHackBlobInfo struct {
```

Comment:
Because the function adds any compatibility hack blob, not just preV5. We are expecting to add more at some point.